node_rate: 20

use_optical_flow_velocities: false  # true -> Using velocity estimates from optical_flow
                                    # false -> Using polled velocity estimates 

mpc_config:
  system_parameters:
    drone:
      mass: 0.320 # [kg]
      aerodynamic_linear_drag_coefficients:
        x: 0.08063504
        y: 0.09929089
        z: 0          # Not estimated

      attitude_control_parameters:  
        time_constants:
          roll:   0.32034106
          pitch:  0.15869484
        input_gains:
          roll:   1.02539815
          pitch:  0.95176013

    gravitational_acceleration: 9.82179 # [m/sÂ²] - for Trondheim at sea level
    
  tuning_parameters:
    nx: 7 
    nu: 4
    m:  1

    objective_function:
      # Standard:  x  y  z    u     v     w     phi   theta   
      # q_diagonal: [2, 2, 1.5, 0.25, 0.25, 0.25, 0.25, 0.25] # [1.25, 1, 1, 0.2, 0.175, 0.1, 0.175, 0.2, 0.0001]  # [2, 2, 1.5, 0.25, 0.25, 0.25, 0.25, 0.25, 0.001] appears very good in sim
      
      # Standard:  x  y  z    u     v     phi   theta   
      q_diagonal: [2, 2, 1.5, 0.25, 0.25, 0.25, 0.25]

      # Augmented: xi   yi   x  y  z    u     v     phi   theta    
      # q_diagonal: [0.1, 0.1, 2, 2, 1.5, 0.25, 0.25, 0.25, 0.25]
      
      r_diagonal: [2, 2, 0, 1]                                      
    
    prediction_horizon: 6
    robust_horizon:     0
    time_step:          0.05  # [s] 

    # Should add the constraints here - future revision

  solving_parameters:
    collocation_degree: 2
    solver_options:
      solver: "mumps"  # MA57 is recommended, https://www.hsl.rl.ac.uk/ipopt/, but some installation is broken
      debug_level: 0
